### ISSUE TITLE
API for downloading/purchasing games in python

### ISSUE THREAD

**Issue by Dogeek** [itchio/itch.io#781]
I'd like to know if there is an API to directly download games, or purchase them, from outside of itch.io. I'm developping the app in python, but I can rewrite the code from another language if necessary.

If there is not, I suggest to create one. Thanks.

**Comment by leafo** [comment]
Could you tell us more about what you're trying to accomplish? Purchases must always be done through our website, but if you're looking for a flexible way to send someone to a page to buy then maybe this is what you want: https://itch.io/docs/api/javascript

We don't have a public API for downloading games, we recommend people to go through our app.

**Comment by Dogeek** [comment]
I'm building an app that is a database for adult games, and some developpers host the games on your website, but not all of them (some use mega.nz, some mediafire and link from their patreon page).

For now, itch.io hosted games are downloaded through your website, but the game won't be downloaded to the app's directory, I'd like to get the version of the downloaded game as well, that's why I was looking for a comprehensive API.

**Comment by leafo** [comment]
I don't think we have any plans right now to publicize a directly download API. You're welcome to investigate how our app works, but you would also need to implement authentication since all downloads require an API key. Using a hard coded API key is not recommended since we may end up blocking it if we see suspicious activity.
